core/ui_events.js

```diff
@@ -113,5 +114,51 @@ Blockly.Events.Click.prototype.fromJson = function(json) {
   this.blockId = json['blockId'];
 };
 
+/**
+ * Class for a trashcan open event.
+ * @param {boolean=} opt_isOpen Whether the trashcan flyout is opening.
+ *    Undefined for a blank event.
+ * @param {string=} opt_workspaceId The workspace identifier for this event.
+ * @extends {Blockly.Events.Ui}
+ * @constructor
+ */
+Blockly.Events.TrashcanOpen = function(opt_isOpen, opt_workspaceId) {
+  Blockly.Events.TrashcanOpen.superClass_.constructor.call(this, opt_workspaceId);
+
+  /**
+   * Whether the trashcan flyout is opening.
+   * @type {boolean|undefined}
+   */
+  this.isOpen = opt_isOpen;
+};
+Blockly.utils.object.inherits(Blockly.Events.TrashcanOpen, Blockly.Events.Ui);
+
+/**
+ * Type of this event.
+ * @type {string}
+ */
+Blockly.Events.TrashcanOpen.prototype.type = Blockly.Events.TRASHCAN_OPEN;
```

Comment:
Does this constant get added in a different PR?

Reply:
Yes